src/core/service.c

```diff
@@ -2008,6 +2015,22 @@ static ServiceState service_determine_dead_state(Service *s) {
         return s->fd_store && s->fd_store_preserve_mode == EXEC_PRESERVE_YES ? SERVICE_DEAD_RESOURCES_PINNED : SERVICE_DEAD;
 }
 
+static void service_set_debug_invocation(Service *s, bool enable) {
+        assert(s);
+
+        if (s->restart_mode != SERVICE_RESTART_MODE_DEBUG)
+                return;
+
+        if (enable == UNIT(s)->debug_invocation)
+                return; /* Nothing to do */
+
+        UNIT(s)->debug_invocation = enable;
+        unit_overwrite_log_level_max(UNIT(s), enable ? LOG_PRI(LOG_DEBUG) : s->exec_context.log_level_max);
```

Comment:
Given this is a Unit field, it is probably better to expose this as `unit_set_debug_invocation`? ` unit_overwrite_log_level_max` can be unexported that way.

Reply:
I don't see the point, it would just be a rename and look the same otherwise